c10/core/DeviceCapability.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <c10/macros/Export.h>
+#include <c10/core/ScalarType.h>
+
+namespace c10 {
+
+#define AT_FORALL_DEVICE_CAPABILITIES(_) \
```

Comment:
PyTorch have provided all supported dypte, please refer to this [link](https://github.com/pytorch/pytorch/blob/e67e3d95f3ab863c8b44a6b16ce9f25d40b0d517/torch/headeronly/core/ScalarType.h#L100) for more informations.

And from my point of view, we need to focus on the generaliztion of implementation, so we need to take all dtype into consideration.